src/psyclone/transformations.py

```diff
@@ -2946,8 +2947,9 @@ def _validate(self, node_list):
             # optimisations are applied. Note that we need to explicitly
             # exclude the OMPParallelDoDirective as it inherits from both
             # OMPDoDirective and OMPParallelDirective.
-            if isinstance(node, (OMPDoDirective, ACCLoopDirective)) and \
-               not isinstance(node, OMPParallelDoDirective):
+            if node.ancestor(OMPParallelDirective,
+                             excluding=[OMPParallelDoDirective]) or \
+                    node.ancestor(ACCParallelDirective):
                 raise TransformationError(
```

Comment:
Strictly speaking, we're now forbidding the insertion of an Extraction region within a (thread) parallel region which seems eminently sensible to me. In fact, I think that means we shouldn't have the `excluding=[...]` bit as that also defines a parallel region. Please could you update and ensure that the error message reflects this.

Reply:
You're right. Updated the constraint and the related test.

Reply:
Great, thanks.